.github/workflows/check-migrations.yml

```diff
@@ -0,0 +1,26 @@
+name: Checks migration files have not been altered.
+
+on:
+  push:
+    branches: "**"
+
+jobs:
+  check_sql_migrations:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - run: git fetch origin main
+      - name: Check for changed SQL migration files
+        run: |
+          # Compare current branch against main branch for modified files
+          CHANGED_FILES=$(git diff --name-status origin/main...HEAD migrations/sql | grep '^M' | cut -f2)
```

Comment:
This won't catch the condition where a developer deletes a migration file.

The migration tool also doesn't care about the name of the file only the prepended number. A user could delete `0...01_nameA.up.sql` and replace it with `0...01_nameB.up.sql` and this will pass.

Reply:
good point on deltion. do you want to allow renamed ones to pass or not?

Reply:
good point, just made the change to fail on deletes and renames

Reply:
This might work?

`git diff --name-status origin/main...HEAD api | grep -E "^M|^D" | cut -f2`